src/main/java/com/gluonhq/substrate/util/FileDeps.java

```diff
@@ -81,22 +82,51 @@
      * Return the path to the JavaFX SDK for this configuration.
      * The path is cached on the provided configuration.
      * If it is not there yet, all dependencies are retrieved.
-     * @param configuration
+     * @param configuration the project configuration
      * @return the location of the JavaFX SDK for the arch-os for this configuration
      * @throws IOException in case anything goes wrong.
      */
-    public static Path getJavaFXSDK(ProjectConfiguration configuration) throws IOException {
-        Path javafxStaticPath = configuration.getJavafxStaticPath();
-        if (Files.exists(javafxStaticPath)) {
-            return javafxStaticPath;
-        }
-        // we don't have the JavaFX SDK yet. setup dependencies, and throw IOException if that "fails"
-        setupDependencies(configuration);
-        if (!Files.exists(javafxStaticPath)) throw new IOException("Fatal error, could not install JavaFX SDK ");
-        return configuration.getJavafxStaticPath();
+    public static Path getJavaSDKPath(ProjectConfiguration configuration) throws IOException {
+        return resolvePath(configuration,
+                configuration.getJavaStaticLibsPath(),
+                "Fatal error, could not install Java SDK ");
     }
 
+    /**
+     * Return the path to the JavaFX SDK for this configuration.
+     * The path is cached on the provided configuration.
+     * If it is not there yet, all dependencies are retrieved.
+     * @param configuration the project configuration
+     * @return the location of the JavaFX SDK for the arch-os for this configuration
+     * @throws IOException in case anything goes wrong.
+     */
+    public static Path getJavaFXSDKLibsPath(ProjectConfiguration configuration) throws IOException {
+        return resolvePath(configuration,
+                configuration.getJavafxStaticLibsPath(),
+                "Fatal error, could not install JavaFX SDK ");
+    }
 
+    /**
+     * Fpr a given path, it verifies that the path exists, or else it tries
+     * to install it. After that, it returns the path for this configuration.
+     * @param configuration the project configuration
+     * @param path the initial path
+     * @param errorMessage a message that will be displayed in case of error
+     * @return the location of the path for this configuration
+     * @throws IOException in case anything goes wrong.
+     */
+    private static Path resolvePath(ProjectConfiguration configuration, Path path, String errorMessage) throws IOException {
+        if (Files.exists(Objects.requireNonNull(path))) {
+            return path;
+        }
+        if (!setupDependencies(Objects.requireNonNull(configuration))) {
+            throw new IOException("Error setting up dependencies");
+        }
+        if (!Files.exists(path)) {
```

Comment:
Checking for existence of the same `path` twice? I seems that we already know it does not exists here

Reply:
Right, `setupDependencies` does the same check. Removed it